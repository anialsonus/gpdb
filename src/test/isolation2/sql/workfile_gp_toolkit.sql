-- This test checks for correct output of gp_toolkit.gp_workfile_entries view
-- It is placed in integration tests because testing for this requires checking
-- for workfiles while a query is still running.

-- check there are no workfiles
1: select segid, prefix, slice from gp_toolkit.gp_workfile_entries order by segid;

1: create table workfile_test(id serial, s text) distributed by (id);
1: insert into workfile_test(s) select v::text from generate_series(1, 1000000) v;

1: select gp_inject_fault_infinite('after_workfile_mgr_create_set', 'skip', dbid) from gp_segment_configuration where content > -1 and role = 'p';

1&: select * from workfile_test t1, workfile_test t2, workfile_test t3, workfile_test t4,
    generate_series(1,12);
-- wait until 1 workfile is created on each segment
2: select gp_wait_until_triggered_fault('after_workfile_mgr_create_set', 1, dbid) from gp_segment_configuration where content > -1 and role = 'p';
-- there should be exactly 3 workfiles, two for each segment (no duplication)
2: select segid, prefix, slice from gp_toolkit.gp_workfile_entries order by segid;

-- interrupt the query
2: select gp_inject_fault('create_function_fail', 'panic', dbid) from gp_segment_configuration where role='p' and content > -1;
2: create function my_function() returns void as $$ begin end; $$ language plpgsql;
1<:
1q:

-- check there are no workfiles left
2: select segid, prefix, slice from gp_toolkit.gp_workfile_entries order by segid;
2: drop table workfile_test;
2q: