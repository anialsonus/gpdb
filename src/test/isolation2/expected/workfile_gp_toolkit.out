-- This test checks for correct output of gp_toolkit.gp_workfile_entries view
-- It is placed in integration tests because testing for this requires checking
-- for workfiles while a query is still running.

-- check there are no workfiles
1: select segid, prefix, slice from gp_toolkit.gp_workfile_entries order by segid;
 segid | prefix | slice 
-------+--------+-------
(0 rows)

1: create table workfile_test(id serial, s text) distributed by (id);
CREATE TABLE
1: insert into workfile_test(s) select v::text from generate_series(1, 1000000) v;
INSERT 0 1000000

1: select gp_inject_fault_infinite('after_workfile_mgr_create_set', 'skip', dbid) from gp_segment_configuration where content > -1 and role = 'p';
 gp_inject_fault_infinite 
--------------------------
 Success:                 
 Success:                 
 Success:                 
(3 rows)

1&: select * from workfile_test t1, workfile_test t2, workfile_test t3, workfile_test t4, generate_series(1,12);  <waiting ...>
-- wait until 1 workfile is created on each segment
2: select gp_wait_until_triggered_fault('after_workfile_mgr_create_set', 1, dbid) from gp_segment_configuration where content > -1 and role = 'p';
 gp_wait_until_triggered_fault 
-------------------------------
 Success:                      
 Success:                      
 Success:                      
(3 rows)
-- there should be exactly 3 workfiles, two for each segment (no duplication)
2: select segid, prefix, slice from gp_toolkit.gp_workfile_entries order by segid;
 segid | prefix              | slice 
-------+---------------------+-------
 0     | slice1_tuplestore_1 | 1     
 1     | slice1_tuplestore_1 | 1     
 2     | slice1_tuplestore_1 | 1     
(3 rows)

-- interrupt the query
2: select gp_inject_fault('create_function_fail', 'panic', dbid) from gp_segment_configuration where role='p' and content > -1;
 gp_inject_fault 
-----------------
 Success:        
 Success:        
 Success:        
(3 rows)
2: create function my_function() returns void as $$ begin end; $$ language plpgsql;
ERROR:  fault triggered, fault name:'create_function_fail' fault type:'panic'  (seg2 10.89.1.2:7004 pid=229746)
1<:  <... completed>
ERROR:  Error on receive from seg2 slice3 10.89.1.2:7004 pid=229725: server closed the connection unexpectedly
	This probably means the server terminated abnormally
	before or while processing the request.
1q: ... <quitting>

-- check there are no workfiles left
2: select segid, prefix, slice from gp_toolkit.gp_workfile_entries order by segid;
 segid | prefix | slice 
-------+--------+-------
(0 rows)
2: drop table workfile_test;
DROP TABLE
2q: ... <quitting>
